TF-6685 Add custom permissions documentation to team project access docs (#371)
hashicorp/terraform-docs-common: website/docs/cloud-docs/api-docs/project-team-access.mdx

  Use the `/team-projects` endpoint to manage team access to a project. List, show, add, update, and remove team access from a project using the HTTP API.
---

> [200]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200

Author: Added to fix the broken status code links and any status code links added later.

## Project Team Access Levels
| Access Level    | Description                                                                                                                                                      |
|-----------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `read`          | Read project and Read workspace access role on project workspaces                                                                                                |
| `write`         | Read project and Write workspace access role on project workspaces                                                                                               |
| `maintain`      | Read project and Admin workspace access role on project workspaces                                                                                               |
| `admin`         | Admin project, Admin workspace access role on project workspaces, create workspaces within project, move workspaces between projects, manage project team access |

> | `custom`        | Custom access permissions on projects and project workspaces.                                                                                                    |

Author: ```suggestion
| `custom`        | Custom access permissions on project and project's workspaces.                                                                                                    |
```

> | `custom`        | Custom access permissions on project and project's workspaces.                                                                                                    |

Reviewer: good catch!

|-----------|------------------------------------------------------------------------------------------------------------------------------------------|
| `:id`     | The ID of the team/project relationship. Obtain this from the [list team access action](#list-team-access-to-a-project) described above. |

-> **Note:** As mentioned in [Add Team Access to a Project](#add-team-access-to-a-project) and [Update Team Access
to a Project](#update-team-access-to-a-project), several permission attributes are not editable unless `access` is
set to `custom`. When access is `read`, `plan`, `write`, or `admin`, these attributes are read-only and reflect the

> implicit permissions granted to the current access level. For example when access is `read`, the implicit

Author: ```suggestion
implicit permissions granted to the current access level. For example, when access is `read`, the implicit
```

| `data.relationships.team.data.id`                         | string  |         | The ID of the team to add to the project.                                                          |
| `data.attributes.project-access.settings`                 | string  | "read"  | If `access` is `custom`, the permission to grant for the project's settings. Can only be used when `access` is `custom`. Valid values include `read`, `update`, or `delete`.                                                   |
| `data.attributes.project-access.teams`                    | string  | "none"  | If `access` is `custom`, the permission to grant for the project's teams. Can only be used when `access` is `custom`. Valid values include `none`, `read`, or `manage`.                                                   |
| `data.attributes.workspace-access.runs`                   | string  | "none"  | If `access` is `custom`, the permission to grant for the project's workspaces' runs. Can only be used when `access` is `custom`. Valid values include `none`, `read`, `plan`, or `apply`.                                                    |

> | `data.attributes.workspace-access.sentinel-mocks`         | string  | "none"  | If `access` is `custom`, the permission to grant for the workspace's Sentinel mocks. Can only be used when `access` is `custom`. Valid values include `none`, or `read`.                                                     |

Author: The S in Sentinel is capitalized -- I double checked on the team access workspace docs.

> | `data.attributes.workspace-access.sentinel-mocks`         | string  | "none"  | If `access` is `custom`, the permission to grant for the workspace's Sentinel mocks. Can only be used when `access` is `custom`. Valid values include `none`, or `read`.                                                     |

Author: ```suggestion
| `data.attributes.workspace-access.sentinel-mocks`         | string  | "none"  | If `access` is `custom`, the permission to grant for the project's workspaces' Sentinel mocks. Can only be used when `access` is `custom`. Valid values include `none`, or `read`.                                                     |
```

> | `data.attributes.workspace-access.state-versions`         | string  | "none"  | If `access` is `custom`, the permission to grant for the workspace's state versions. Can only be used when `access` is `custom`. Valid values include `none`, `read-outputs`, `read`, or `write`.                                                    |

Author: ```suggestion
| `data.attributes.workspace-access.state-versions`         | string  | "none"  | If `access` is `custom`, the permission to grant for the project's workspaces state versions. Can only be used when `access` is `custom`. Valid values include `none`, `read-outputs`, `read`, or `write`.                                                    |
```

| `data.attributes.workspace-access.variables`              | string  | "none"  | If `access` is `custom`, the permission to grant for the project's workspaces' variables. Can only be used when `access` is `custom`. Valid values include `none`, `read`, or `write`.                                                    |
| `data.attributes.workspace-access.create`                 | boolean | false  | If `access` is `custom`, this permission allows the team to create workspaces in the project.                 |
| `data.attributes.workspace-access.locking`                | boolean | false  | If `access` is `custom`, the permission granting the ability to manually lock or unlock the project's workspaces. Can only be used when `access` is `custom`.                                                   |

> | `data.attributes.workspace-access.move`                   | boolean | false  | If `access` is `move`, this permission allows the team to move workspaces into and out of the project.        |

Author: ```suggestion
| `data.attributes.workspace-access.move`                   | boolean | false  | If `access` is `move`, this permission allows the team to move workspaces into and out of the project. The team must also have permissions to the project(s) receiving the the workspace(s).       |
```

| Permissions                         | `read` | `write` | `maintain` | `admin `   | `custom` default |
|-------------------------------------|--------|---------|------------|----------|--------------------|
| project-access.settings             | "read" | "read"  | "read"     | "delete" | "read"             |
| project-access.teams                | "none" | "none"  | "none"     | "manage" | "none"             |
| workspace-access.runs               | "read" | "apply" | "apply"    | "apply"  | "none"             |
| workspace-access.sentinel-mocks"    | "none" | "read"  | "read"     | "read"   | "none"             |
| workspace-access.state-versions     | "read" | "write" | "write"    | "write"  | "none"             |
| workspace-access.variables          | "read" | "write" | "write"    | "write"  | "none"             |
| workspace-access.create             | false  | false   | true       | true     | false              |
| workspace-access.locking            | false  | true    | true       | true     | false              |
| workspace-access.move               | false  | false   | false      | true     | false              |

> | workspace-access.run-tasks          | false  | false   | true       | true     | false              |

Reviewer 2: Beautiful!! 👏

  --request DELETE \
  https://app.terraform.io/api/v2/team-projects/tprj-WbG7p5KnT7S7HZqw
```

## Implied Custom Permission Levels

> As mentioned above, when setting team access levels of `read`, `write`, `maintain`, or `admin`, the following permissions that can be set individually when using an access level of `custom`, are given values based on the access level.

Reviewer 3: Oops, I skipped this file in my first review!

I suggest shortening to (I think this still covers the same information):
```suggestion
As mentioned above, when setting team access levels (`read`, `write`, `maintain`, or `admin`), you can individually set the following permissions if you use the `custom` access level.
```

> Below is each access level, and the implicitly granted custom permission. When using `custom` as the access levels, any permission not specified will be given the default level listed

Reviewer 3: ```suggestion
The below table lists each access level alongside its implicit custom permission level.  If you use the `custom` access level and do not specify a certain permission's level, that permission uses the default value listed below.
```
